### PR TITLE
test: increase test coverage for pkg/pusher

### PR DESCRIPTION
**Pull request by tpresa**
## Summary
- Increased test coverage for `pkg/pusher` from 41.4% to 91.4%
- Added comprehensive tests for previously untested code paths

## Related Issue
Part of #30814 - improving test coverage across the codebase

## Test coverage improvements
- Added tests for `newRegistryClient()` function with various TLS configurations
- Added tests for `push()` method including chart loading and provenance file handling
- Added tests for error scenarios and edge cases

## Test results
```
ok  	helm.sh/helm/v4/pkg/pusher	10.699s	coverage: 91.4% of statements
```

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

**Comment by robertsirc** [comment]
Hello, thank you for your pull request as we do appreciate the help we cannot review this PR until you fix your DCO.

File: pkg/pusher/ocipusher_test.go [hunk 1]
```diff
@@ -94,3 +97,329 @@ func TestNewOCIPusher(t *testing.T) {
...
+					t.Fatal(err)
+				}
+				
+				if _, err := io.Copy(dst, src); err != nil {
+					t.Fatal(err)
+				}
+				dst.Close()
+				
+				// Make the file unreadable
+				if err := os.Chmod(tempChart, 0000); err != nil {
```
Comment (benoittgt) [inline pkg/pusher/ocipusher_test.go:304]: Does it mean we should flag the file with `//go:build !windows` like in https://github.com/helm/helm/blob/main/pkg/helmpath/home_unix_test.go#L14 ?

Comment (TerryHowe) [reply pkg/pusher/ocipusher_test.go:304]: Cool, I didn't know you could do `!` on that

Comment (tpresa) [reply pkg/pusher/ocipusher_test.go:304]: Yes, thank you! Done.

**Review by robertsirc — APPROVED** [review]
LGTM

**Review by mattfarina — APPROVED** [review]
lgtm
